Requestify computation of break/continue targets

hamishknight opened the pull request:
Introduce BreakTargetRequest and ContinueTargetRequest that perform the ASTScope lookup for the target of a break or continue.

hamishknight commented:
@swift-ci please test

hamishknight commented:
@swift-ci please test Windows

rintaro commented on include/swift/AST/Stmt.h:
```diff
@@ -1336,13 +1336,13 @@ class BreakStmt : public Stmt {
   Identifier TargetName; // Named target statement, if specified in the source.
   SourceLoc TargetLoc;
-  LabeledStmt *Target = nullptr;  // Target stmt, wired up by Sema.
+  DeclContext *DC;
```
Could you add a forward declaration for `DeclContext` in this `.h` file?
As far as I can see. No header in this file has `#include "swift/AST/DeclContext.h"`

hamishknight replied:
Good catch!

hamishknight committed:
Requestify computation of break/continue targets

Introduce BreakTargetRequest and
ContinueTargetRequest that perform the ASTScope
lookup for the target of a break or continue.

hamishknight commented:
@swift-ci please smoke test

hamishknight commented:
@swift-ci please smoke test macOS

hamishknight commented:
@swift-ci please smoke test macOS

hamishknight commented:
@swift-ci please smoke test macOS
